File: problems/228-Summary-Ranges/solution.go
func summaryRanges(nums []int) []string {
	result := make([]string, 0)

	if len(nums) == 0 {
		return result
	}

	start, end := nums[0], nums[0]
	for i := 1; i < len(nums); i++ {
		if nums[i] > end+1 {
			if start < end {
				result = append(result, fmt.Sprintf("%d->%d", start, end))
			} else {
				result = append(result, fmt.Sprintf("%d", end))
			}
			start = nums[i]
			end = nums[i]
		} else {
			end = nums[i]
		}
	}

	if start < end {
		result = append(result, fmt.Sprintf("%d->%d", start, end))
	} else {
		result = append(result, fmt.Sprintf("%d", end))
	}

	return result
}

File: problems/228-Summary-Ranges/solution.java
class Solution {
    public List<String> summaryRanges(int[] nums) {
        ArrayList<String> result = new ArrayList<>();

        if (nums.length == 0) {
            return result;
        }

        int start = nums[0], prev = nums[0];
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == prev + 1) {
                prev++;
                continue;
            }

            if (start == prev) {
                result.add(String.format("%d", start));
            } else {
                result.add(String.format("%d->%d", start, prev));
            }

            start = nums[i];
            prev = nums[i];
        }

        if (start == prev) {
            result.add(String.format("%d", start));
        } else {
            result.add(String.format("%d->%d", start, prev));
        }

        return result;
    }
}

